crates/rpc/rpc-types-compat/src/transaction.rs

```diff
@@ -81,19 +82,83 @@ pub trait TransactionCompat: Send + Sync + Unpin + Clone + Debug {
     ) -> Result<Self::TxEnv, Self::Error>;
 }
 
-/// Converts `self` into `T`.
+/// Converts `self` into `T`. The opposite of [`FromConsensusTx`].
 ///
 /// Should create an RPC transaction response object based on a consensus transaction, its signer
-/// [`Address`] and an additional context.
+/// [`Address`] and an additional context [`IntoRpcTx::TxInfo`].
+///
+/// Avoid implementing [`IntoRpcTx`] and use [`FromConsensusTx`] instead. Implementing it
+/// automatically provides an implementation of [`IntoRpcTx`] thanks to the blanket implementation
+/// in this crate.
+///
+/// Prefer using [`IntoRpcTx`] over [`FromConsensusTx`] when specifying trait bounds on a generic
+/// function to ensure that types that only implement [`IntoRpcTx`] can be used as well.
 pub trait IntoRpcTx<T> {
     /// An additional context, usually [`TransactionInfo`] in a wrapper that carries some
     /// implementation specific extra information.
     type TxInfo;
 
-    /// Performs the conversion.
+    /// Performs the conversion consuming `self` with `signer` and `tx_info`. See [`IntoRpcTx`]
+    /// for details.
     fn into_rpc_tx(self, signer: Address, tx_info: Self::TxInfo) -> T;
 }
 
+/// Converts `T` into `self`. It is reciprocal of [`IntoRpcTx`].
+///
+/// Should create an RPC transaction response object based on a consensus transaction, its signer
+/// [`Address`] and an additional context [`FromConsensusTx::TxInfo`].
+///
+/// Prefer implementing [`FromConsensusTx`] over [`IntoRpcTx`] because it automatically provides an
+/// implementation of [`IntoRpcTx`] thanks to the blanket implementation in this crate.
+///
+/// Prefer using [`IntoRpcTx`] over using [`FromConsensusTx`] when specifying trait bounds on a
+/// generic function. This way, types that directly implement [`IntoRpcTx`] can be used as arguments
+/// as well.
+pub trait FromConsensusTx<T> {
+    /// An additional context, usually [`TransactionInfo`] in a wrapper that carries some
+    /// implementation specific extra information.
+    type TxInfo;
+
+    /// Performs the conversion consuming `tx` with `signer` and `tx_info`. See [`FromConsensusTx`]
+    /// for details.
+    fn from_consensus_tx(tx: T, signer: Address, tx_info: Self::TxInfo) -> Self;
```

Comment:
this makes sense to me, this should also allow multiple conversions into the same rpc type but from different consensus types